flowey/flowey_lib_common/src/publish_test_results.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Publish test results.
+//!
+//! - On ADO, this will hook into the backend's native JUnit handling.
+//! - On Github, this will publish artifacts containing the raw JUnit XML file
+//!   and any optional attachments.
+//! - When running locally, this will optionally copy the XML files and any
+//!   attachments to the provided artifact directory.
+
+use crate::_util::copy_dir_all;
+use flowey::node::prelude::*;
+use std::collections::BTreeMap;
+
+flowey_request! {
+    pub struct Request {
+        /// Path to a junit.xml file
+        ///
+        /// HACK: this is an optional since `flowey` doesn't (yet?) have any way
+        /// to perform conditional-requests, and there are instances where nodes
+        /// will only conditionally output JUnit XML.
+        ///
+        /// To keep making forward progress, I've tweaked this node to accept an
+        /// optional... but this ain't great.
+        pub junit_xml: ReadVar<Option<PathBuf>>,
+        /// Brief string used when publishing the test.
+        /// Must be unique to the pipeline.
+        pub test_label: String,
+        /// Additional files or directories to upload
+        /// (not used on ADO unless the associated boolean is true).
```

Comment:
You're conflating the current implementation with the desired long-term intent of this API.

I would rephrase this along the lines of "the boolean indicates whether this attachment is referred to by the JUnit XML. On backends with native junit attachment support, these attachments will not be uploaded as distinct artifacts, and will instead be uploaded through the junit integration"